core/src/main/scala/producer/KafkaProducerConfig.scala

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2017 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package freestyle
+package kafka
+
+import org.apache.kafka.clients.producer.KafkaProducer
+import org.apache.kafka.common.serialization.Serializer
+import collection.JavaConverters._
+
+case class KafkaProducerConfig[K, V](
+    configs: Map[String, Any],
+    keyValueSerializers: Option[(Serializer[K], Serializer[V])])
+    extends UnderlyingKafkaProducer[K, V] {
+  override def producer: KafkaProducer[K, V] = KafkaProducerConfig.producerFromConfig(this)
+}
+
+object KafkaProducerConfig {
+
+  private def toAnyRefMap(m: Map[String, Any]): java.util.Map[String, AnyRef] =
+    m.asInstanceOf[Map[String, AnyRef]].asJava
```

Comment:
What would happen for those configurations that have an integer or boolean as a value? In that case, is safe going from AnyVal to AnyRef?